### PR TITLE
chore: Fix linting errors found by shellcheck

### PR DESCRIPTION
**Pull request by rapphil**
**Description:** Fix linting errors found by shellcheck

Also run linting on all shell scripts on the repository. We are assuming that all shell scripts have the `.sh` suffix.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: tools/release/image-binary-release/delete-s3-release.sh [hunk 1]
```diff
@@ -84,14 +84,14 @@ function delete_s3_objects_from_s3_bucket() {
     )
 
     for i in "${s3_path[@]}"; do
-        s3_latest_key=$(echo "${i}")
-        s3_version_key=$(echo ${s3_latest_key} | sed s/latest/${version}/g)
+        s3_latest_key="${i}"
+        s3_version_key="${s3_latest_key/latest/${version}}"
```
Comment (bryan-aguilar) [inline tools/release/image-binary-release/delete-s3-release.sh:88]: Can you explain this change a bit more?

Comment (rapphil) [reply tools/release/image-binary-release/delete-s3-release.sh:88]: https://www.gnu.org/software/bash/manual/bash.html#Shell-Parameter-Expansion

Comment (rapphil) [reply tools/release/image-binary-release/delete-s3-release.sh:88]: This feature: `${parameter/pattern/string}`

Comment (rapphil) [reply tools/release/image-binary-release/delete-s3-release.sh:88]: This was the warning that I was getting: https://www.shellcheck.net/wiki/SC2001

Comment (bryan-aguilar) [reply tools/release/image-binary-release/delete-s3-release.sh:88]:
```suggestion
        s3_version_key="${s3_latest_key//latest/${version}}"
```
Should it be two slashes here based on the link you provided?

**Review by bryan-aguilar — APPROVED** [review]

**Pull request merged**